snuba/datasets/plans/single_storage.py

```diff
@@ -18,7 +19,16 @@
 
 
 class SimpleQueryPlanExecutionStrategy(QueryPlanExecutionStrategy):
+    def __init__(
+        self, splitters: Optional[Sequence[StorageQuerySplitStrategy]] = None
+    ) -> None:
+        self.__splitters = splitters or []
+
     def execute(self, request: Request, runner: QueryRunner) -> RawQueryResult:
+        for splitter in self.__splitters:
+            if splitter.can_execute(request):
+                return splitter.execute(request, runner)
```

Comment:
Would it make sense to skip `can_execute` and directly call the splitter (to avoid things like the double request copy in `ColumnSplitQueryStrategy`), letting it return `None` it if it is unable to run the query, indicating that another strategy should be used?

Reply:
Yes it is possible, but is it better? Feels a little implicit that we call `execute` and depending on the result we figure out that nothing was executed and we have to try again with another splitter. Anyway, not a strong opinion.

Reply:
Other than potentially (and admittedly minor) efficiency improvements, the other benefit would be that it reduces the potential for this precondition to be violated: https://github.com/getsentry/snuba/pull/865#discussion_r404552098 Once that type becomes immutable (at least via the type definition — assuming it will) that's less of a concern from my perspective. I think either approach right now has it's pros and cons, I'll leave the decision up to you.

Reply:
good point, mutability can be an issue, will make the change.